### PR TITLE
[perf] Reduce eval local to remote tracking latency by caching the arm token

### PR DESCRIPTION
**Pull request by ninghu**
# Description

Based on the investigation, local to remote tracking involves multiple client to service calls. Each call requires acquiring an ARM token from AAD, with each token acquisition taking about 2 seconds. By caching the token, we could reduce the end-to-end time of the evaluate API call with one evaluator from 76 seconds to 51 seconds, achieving around a 30% improvement.

For more details, please check out [here](https://microsoft-my.sharepoint.com/:w:/p/ninhu/ETB_zdMkFrdAuf3Lcg9ssrUB6RVmyuFs5Un1G74O1HlwSA?e=cBVmsw)

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-azure/promptflow/azure/_utils/_token_cache.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import time
```
Comment (wangchao1230) [inline src/promptflow-azure/promptflow/azure/_utils/_token_cache.py:1]: missed a copy right header of this file

File: src/promptflow-azure/promptflow/azure/_utils/_token_cache.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+class ArmTokenCache(metaclass=SingletonMeta):
+    DEFAULT_TTL_SECS = 1800
+
+    def __init__(self):
+        self._ttl_secs = self.DEFAULT_TTL_SECS
+        self._cache = {}
+
+    def _is_token_valid(self, entry):
+        return time.time() < entry["expires_at"]
```
Comment (wangchao1230) [inline src/promptflow-azure/promptflow/azure/_utils/_token_cache.py:24]: should we do refresh for token which is about to expire in near time?

azure-identity will do it for 300 seconds
https://github.com/Azure/azure-sdk-for-python/blob/9fb4c52d988ceb3b9583b2dff48d2e0304228590/sdk/identity/azure-identity/azure/identity/_internal/shared_token_cache.py#L240

File: src/promptflow-azure/promptflow/azure/_utils/general.py [hunk 3]
```diff
@@ -6,6 +6,8 @@
 
 from promptflow.core._connection_provider._utils import get_arm_token, get_token
 
+from ._token_cache import ArmTokenCache
```
Comment (0mza987) [inline src/promptflow-azure/promptflow/azure/_utils/general.py:9]: NIT: Use absolute import to align with other imports.

File: src/promptflow-azure/promptflow/azure/_utils/_token_cache.py (hunk 2 above)
Comment (ninghu) [reply src/promptflow-azure/promptflow/azure/_utils/_token_cache.py]: I've updated the code to align with azure-identity's behavior.

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**